client_src/js/lib/modules/printerSettings.js

```diff
@@ -1,1325 +1,1403 @@
-    elements.connectPage.filesClean.innerHTML = `<i class="fas fa-server"></i> <b>File Clean</b><br><b>Last Checked: </b>${Calc.dateClean(
-      printer.systemChecks.cleaning.file.date
+    pageElements.connectPage.apiClean.className = `btn btn-${currentPrinter.systemChecks.cleaning.information.status} mb-1 btn-block`;
+    pageElements.connectPage.filesClean.innerHTML = `<i class="fas fa-server"></i> <b>File Clean</b><br><b>Last Checked: </b>${Calc.dateClean(
+      currentPrinter.systemChecks.cleaning.file.date
     )}`;
-    elements.connectPage.filesClean.className = `btn btn-${printer.systemChecks.cleaning.file.status} mb-1 btn-block`;
-    elements.connectPage.stateClean.innerHTML = `<i class="fas fa-server"></i> <b>Job Clean</b><br><b>Last Checked: </b>${Calc.dateClean(
-      printer.systemChecks.cleaning.job.date
+    pageElements.connectPage.filesClean.className = `btn btn-${currentPrinter.systemChecks.cleaning.file.status} mb-1 btn-block`;
+    pageElements.connectPage.stateClean.innerHTML = `<i class="fas fa-server"></i> <b>Job Clean</b><br><b>Last Checked: </b>${Calc.dateClean(
+      currentPrinter.systemChecks.cleaning.job.date
     )}`;
-    elements.connectPage.stateClean.className = `btn btn-${printer.systemChecks.cleaning.job.status} mb-1 btn-block`;
+    pageElements.connectPage.stateClean.className = `btn btn-${currentPrinter.systemChecks.cleaning.job.status} mb-1 btn-block`;
 
-    if (printer.printerState.colour.category === "Offline") {
-      const printerProfileBtn = document.getElementById("printer-profile-btn");
-      const printerGcodeBtn = document.getElementById("printer-gcode-btn");
-      const printerOtherSettings = document.getElementById(
-        "printer-settings-btn"
-      );
-      printerProfileBtn.disabled = true;
-      printerGcodeBtn.disabled = true;
-      printerOtherSettings.disabled = true;
-      if (!printerProfileBtn.classList.contains("notyet")) {
-        printerProfileBtn.classList.add("notyet");
+    if (!printerOnline) {
+      pageElements.menu.printerProfileBtn.disabled = true;
+      pageElements.menu.printerGcodeBtn.disabled = true;
+      pageElements.menu.printerOtherSettings.disabled = true;
+
+      if (!pageElements.menu.printerProfileBtn.classList.contains("notyet")) {
+        pageElements.menu.printerProfileBtn.classList.add("notyet");
       }
-      if (!printerGcodeBtn.classList.contains("notyet")) {
-        printerGcodeBtn.classList.add("notyet");
+      if (!pageElements.menu.printerGcodeBtn.classList.contains("notyet")) {
+        pageElements.menu.printerGcodeBtn.classList.add("notyet");
       }
-      if (!printerOtherSettings.classList.contains("notyet")) {
-        printerOtherSettings.classList.add("notyet");
+      if (
+        !pageElements.menu.printerOtherSettings.classList.contains("notyet")
+      ) {
+        pageElements.menu.printerOtherSettings.classList.add("notyet");
```

Comment:
If the printer goes offline whilst your on this page, this will make the UI react.